src/pdf/pdf-view.js

```diff
@@ -1671,7 +1671,19 @@ class PDFView {
 				if (this._selectedAnnotationIDs.length !== 0) {
 					this._onSelectAnnotations([], event);
 				}
-				this._onOpenViewContextMenu({ x: br.x + event.clientX, y: br.y + event.clientY });
+				// If this is a keyboard contextmenu event, its position won't take our
+				// text selection into account since we don't use browser selection APIs.
+				// Position the menu manually.
+				if (event.mozInputSource === 6 && this._selectionRanges.length) {
+					let selectionBoundingRect = getPositionBoundingRect(this._selectionRanges[0].position);
+					this._onOpenViewContextMenu({
+						x: br.x + selectionBoundingRect.right,
+						y: br.y + selectionBoundingRect.bottom
```

Comment:
This won't position the context menu properly because `selectionBoundingRect` is only a rect ([x1, y1, x2, y2]). But I can take care of this code.

Reply:
So we can use this code for that:
```js
if (event.mozInputSource === 6 && this._selectionRanges.length) {
  const EXTRA_VERTICAL_PADDING = 10;
  let selectionBoundingRect = this.getClientRectForPopup(this._selectionRanges[0].position);
  this._onOpenViewContextMenu({
  x: br.x + selectionBoundingRect[0],
  y: br.y + selectionBoundingRect[3] + EXTRA_VERTICAL_PADDING
  });
}
```

And that should be enough if we don't want to position other context menus i.e. annotation.